url80587fc7d5_dennervidal_CloudSim_DVFS_tgz-pJ8-Examples_Dvfs_example_simpleJ8/src/org/cloudbus/cloudsim/power/PowerDatacenterNonPowerAware.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Title:        CloudSim Toolkit
+ * Description:  CloudSim (Cloud Simulation) Toolkit for Modeling and Simulation of Clouds
+ * Licence:      GPL - http://www.gnu.org/copyleft/gpl.html
+ *
+ * Copyright (c) 2009-2012, The University of Melbourne, Australia
+ */
+
+package org.cloudbus.cloudsim.power;
+
+import java.util.List;
+import java.util.Map;
+
+import org.cloudbus.cloudsim.DatacenterCharacteristics;
+import org.cloudbus.cloudsim.Log;
+import org.cloudbus.cloudsim.Storage;
+import org.cloudbus.cloudsim.Vm;
+import org.cloudbus.cloudsim.VmAllocationPolicy;
+import org.cloudbus.cloudsim.core.CloudSim;
+import org.cloudbus.cloudsim.core.CloudSimTags;
+import org.cloudbus.cloudsim.core.predicates.PredicateType;
+
+/**
+ * PowerDatacenterNonPowerAware is a class that represents a non-power aware data center in the
+ * context of power-aware simulations.
+ * 
+ * If you are using any algorithms, policies or workload included in the power package please cite
+ * the following paper:
+ * 
+ * Anton Beloglazov, and Rajkumar Buyya, "Optimal Online Deterministic Algorithms and Adaptive
+ * Heuristics for Energy and Performance Efficient Dynamic Consolidation of Virtual Machines in
+ * Cloud Data Centers", Concurrency and Computation: Practice and Experience, ISSN: 1532-0626, Wiley
+ * Press, New York, USA, 2011, DOI: 10.1002/cpe.1867
+ * 
+ * @author Anton Beloglazov
+ * @since CloudSim Toolkit 2.0
+ */
+public class PowerDatacenterNonPowerAware extends PowerDatacenter {
+
+	/**
+	 * Instantiates a new datacenter.
+	 * 
+	 * @param name the name
+	 * @param characteristics the res config
+	 * @param schedulingInterval the scheduling interval
+	 * @param utilizationBound the utilization bound
+	 * @param vmAllocationPolicy the vm provisioner
+	 * @param storageList the storage list
+	 * 
+	 * @throws Exception the exception
+	 */
+	public PowerDatacenterNonPowerAware(
+			String name,
+			DatacenterCharacteristics characteristics,
+			VmAllocationPolicy vmAllocationPolicy,
+			List<Storage> storageList,
+			double schedulingInterval) throws Exception {
+		super(name, characteristics, vmAllocationPolicy, storageList, schedulingInterval);
+	}
+
+	/**
+	 * Updates processing of each cloudlet running in this PowerDatacenter. It is necessary because
+	 * Hosts and VirtualMachines are simple objects, not entities. So, they don't receive events and
+	 * updating cloudlets inside them must be called from the outside.
+	 * 
+	 * @pre $none
+	 * @post $none
+	 */
+	@Override
+	protected void updateCloudletProcessing() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.